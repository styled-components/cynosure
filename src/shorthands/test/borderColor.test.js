// @flow
import borderColor from '../borderColor'

describe('borderColor', () => {
  it('properly applies a value when passed only one', () => {
    expect(borderColor('red')).toEqual({
      borderBottomColor: 'red',
      borderLeftColor: 'red',
      borderRightColor: 'red',
      borderTopColor: 'red',
    })
  })
  it('properly applies values when passed two', () => {
    expect(borderColor('red', 'blue')).toEqual({
      borderBottomColor: 'red',
      borderLeftColor: 'blue',
      borderRightColor: 'blue',
      borderTopColor: 'red',
    })
  })
  it('properly applies values when passed three', () => {
    expect(borderColor('red', 'blue', 'green')).toEqual({
      borderBottomColor: 'green',
      borderLeftColor: 'blue',
      borderRightColor: 'blue',
      borderTopColor: 'red',
    })
  })
  it('properly applies values when passed four', () => {
    expect(borderColor('red', 'blue', 'green', 'yellow')).toEqual({
      borderBottomColor: 'green',
      borderLeftColor: 'yellow',
      borderRightColor: 'blue',
      borderTopColor: 'red',
    })
  })
})
