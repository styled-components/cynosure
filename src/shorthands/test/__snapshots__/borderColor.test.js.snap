exports[`borderColor properly applies a value when passed only one 1`] = `
Object {
  "borderBottomColor": "red",
  "borderLeftColor": "red",
  "borderRightColor": "red",
  "borderTopColor": "red",
}
`;

exports[`borderColor properly applies values when passed four 1`] = `
Object {
  "borderBottomColor": "green",
  "borderLeftColor": "yellow",
  "borderRightColor": "blue",
  "borderTopColor": "red",
}
`;

exports[`borderColor properly applies values when passed three 1`] = `
Object {
  "borderBottomColor": "green",
  "borderLeftColor": "blue",
  "borderRightColor": "blue",
  "borderTopColor": "red",
}
`;

exports[`borderColor properly applies values when passed two 1`] = `
Object {
  "borderBottomColor": "red",
  "borderLeftColor": "blue",
  "borderRightColor": "blue",
  "borderTopColor": "red",
}
`;
